services/supervisor/src/transactions.rs

```diff
@@ -169,17 +169,23 @@ where
             return Err(Error::ActualFromIsPast.into());
         }
 
+        let mut schema = Schema::new(context.service_data());
+
         // Verifies that there are no pending config changes.
-        if Schema::new(context.service_data())
-            .pending_proposal
-            .exists()
-        {
-            return Err(Error::ConfigProposeExists.into());
+        if let Some(proposal) = schema.pending_proposal.get() {
+            // We have a proposal, check that it's actual.
+            if current_height < proposal.config_propose.actual_from {
+                return Err(Error::ConfigProposeExists.into());
+            } else {
+                // Proposal is outdated but was not removed (e.g. because of the panic
+                // during config applying), clean it.
+                schema.pending_proposal.remove();
+            }
         }
+        drop(schema);
```

Comment:
What about additional scope ? The question is: what is more idiomatic approach here? @alekseysidorov what do you think ? I'd avoid explicit call of `drop`.

Reply:
I'm not a fan of it either, but since `drop` was [already introduced](https://github.com/exonum/exonum/blob/dynamic_services/services/supervisor/src/lib.rs#L209) I decided to match the style.

Reply:
Ok. I got it.